docs_src/design/adr/device-service/0012-DeviceService-Filters.md

```diff
@@ -0,0 +1,152 @@
+# Device Service Filters
+
+## Status
+
+### Proposed
+
+- design for Hanoi
+- implementation for Ireland
+
+## Context
+
+In EdgeX today, sensor/device data collected can be "filtered" by [application services](../../../microservices/application/ApplicationServices.md) before being exported or sent to some [north side](../../../general/Definitions.md#south-and-north-side) application or system. Built-in application service functions (available through the app services SDK) allow EdgeX event/reading objects to be filtered by device name or by device resource name.  That is, event/readings can be filtered by:
+
+- which device sent the event/reading (as determined by the Event device property).
+- the classification or origin (such as temperature or humidity) of data produced by the device as determined by the Reading's name property (which used to be the value descriptor and now refers to the device resource name).
+
+### Two Levels of Device Service Filtering
+
+There are potentially two places where "filtering" in a device service could be useful.  
+
+- One (Sensor Data Filter) - after the device service has communicated with the sensor or device to get sensor values (but before the service creates `Event/Reading` objects and pushes those to core data).  A sensor data filter would allow the device service to essentially ignore some of the raw sensed data.  This would allow for some device service optimization in that the device service would not have perform type transformations and creation of event/reading objects if the data can be eliminated at this early stage.  This first level filtering would, **if put in place**, likely occur in code associated with the read command gets done by the `ProtocolDriver`.
+- Two (Reading Filter) - after the sensor data has been collected and read and put into `Event/Reading` objects, there is a desire to filter some of the `Readings` based on the `Reading` values or `Reading` name (which is the device resource name and formerly the value descriptor name) or some combination of value and name.
+
+At this time, **this design only addresses the need for the second filter (Reading Filter)**.  At the time of this writing, no applicable use case has yet to be defined to warrant the Sensor Data Filter.
+
+### Reading Filters
+Reading filters will allow, not unlike application service filter functions today, to have `Readings` in an `Event` to be removed if:
+
+- the value was outside or inside some range, or the value was greater than, less than or equal to some value
+    - based on the `Reading` value (numeric) of a `Reading` outside a specified range (min/max) described in the device profile for a device resource.  Thus avoiding sending in outlier or jittery data `Readings` that could negatively effect analytics.
```

Comment:
This says that range (Min/Max) being used to filter is described in a device profile's device resource, yet what's being proposed is that the range be defined via service configuration?

Reply:
based on earlier draft where we considered the profile the placement of value filters.  Good catch and updated accordingly.